service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java

```
@@ -827,4 +836,29 @@ public Map<String, Object> unifyAttributeMaps(
         }
         return existing;
     }
+
+    private void setDocumentsFromPendingToNew(Map<UUID, String> customerProvidedDocumentsMap) {
+        for (Map.Entry<UUID, String> entry : customerProvidedDocumentsMap.entrySet()) {
+            UUID key = entry.getKey();
+
+            Optional<CustomerProvidedDocument> optionalCustomerProvidedDocument =
+                    customerProvidedDocumentRepository.findById(key);
+            if (optionalCustomerProvidedDocument.isPresent()) {
+                CustomerProvidedDocument customerProvidedDocument =
+                        optionalCustomerProvidedDocument.get();
+                if (!customerProvidedDocument
+                        .getDataPath()
+                        .equals(customerProvidedDocumentsMap.get(key))) {
+                    throw new BusinessLogicException(
```

Not sure that we want to throw an exception here. While I suspect this won't happen under normal usage, it could happen after we implement version management for configuration and include features for version migration of live transactions.

I think it would be better to update the path attribute of the `CustomerProvidedDocument` in this case.